src/openag_brain/software_modules/recipe_handler.py

```diff
@@ -12,248 +12,276 @@
 instance of this module per environment in the system.
 """
 
-import sys
 import time
 import rospy
-import random
 from openag.db_names import ENVIRONMENTAL_DATA_POINT, RECIPE
 from openag.cli.config import config as cli_config
 from openag.models import EnvironmentalDataPoint
 from openag.var_types import RECIPE_START, RECIPE_END, EnvVar
 from couchdb import Server
 from std_msgs.msg import Float64
-from threading import Event, Timer
-
+from threading import RLock
 from openag_brain import params, services
 from openag_brain.srv import StartRecipe, Empty
+from openag_brain.utils import gen_doc_id
+from openag_brain.memoize import memoize
+from openag_brain.multidispatch import multidispatch
 
-class Recipe(object):
-    def __init__(self, _id, operations, start_time):
-        self._id = _id
-        self.operations = operations
-        self.start_time = start_time
+# Create a tuple constant of valid environmental variables
+VALID_VARIABLES = tuple(EnvVar.items.keys())
```

Comment:
Since this tuple is only used for lookups, using a set instead would be more efficient. Or a frozenset if you want something immutable.